### PR TITLE
[ETP-860] - Enhancement - Add ticket restock functionality upon attendee deletion for TicketsCommerce 

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[ETP-860] 

### 🗒️ Description

* Properly restock Tickets Commerce tickets after deleting corresponding attendee.

### 🎥 Artifacts 
🎥 screencast(s): https://share.cleanshot.com/4Z4m7k5w

### ✔️ Checklist
- [x] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ETP-860]: https://theeventscalendar.atlassian.net/browse/ETP-860?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Module.php [hunk 1]
```diff
@@ -604,16 +606,28 @@ public function delete_ticket( $event_id, $ticket_id ) {
...
+		}
 
-		return $deleted;
+		if ( $deleted ) {
+			// Run anything we might need on parent method.
+			parent::delete_ticket( $event_id, $ticket_id );
+			return true;
+		}
+		
+		return false;
```
Comment (codingmusician) [inline src/Tickets/Commerce/Module.php:630]: Bail early if false
```suggestion
		if ( ! $deleted ) {
			return false;
		}
		
		// Run anything we might need on parent method.
		parent::delete_ticket( $event_id, $ticket_id );
		return true;
```

File: tests/commerce_integration/TEC/Tickets/Commerce/Stock/UpdateTest.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		$attendees = tribe_attendees()->where( 'event_id', $event_id )->all();
+
+		$this->assertEquals( 2, count( $attendees ), 'There should be 2 attendees' );
+
+		// delete attendee.
+		$attendee = $attendees[0];
+		$deleted = tribe( Attendee::class )->delete( $attendee->ID );
+
+		$new_count = tec_tc_attendees()->by( 'event_id', $event_id )->count();
+		$this->assertEquals( 1, $new_count, 'There should be 4 attendees' );
```
Comment (codingmusician) [inline tests/commerce_integration/TEC/Tickets/Commerce/Stock/UpdateTest.php:50]:
```suggestion
		$this->assertEquals( 1, $new_count, 'There should be 1 attendee' );
```

File: src/Tickets/Commerce/Hooks.php [hunk 3]
```diff
@@ -74,7 +74,7 @@ protected function add_actions() {
 
 		add_action( 'tribe_events_tickets_attendees_event_details_top', [ $this, 'setup_attendance_totals' ] );
 		add_action( 'trashed_post', [ $this, 'maybe_redirect_to_attendees_report' ] );
-		add_action( 'tickets_tpp_ticket_deleted', [ $this, 'update_stock_after_deletion' ], 10, 3 );
+		add_action( 'tec_tickets_commerce_attendee_before_delete', [ $this, 'update_stock_after_attendee_deletion' ] );
```
[the remark this answers is not in the document]
Comment (rafsuntaskin) [reply src/Tickets/Commerce/Hooks.php:77]: These are Tickets Commerce hooks only, so won't impact any TribeCommerce users. It was here without any impact.

File: src/Tickets/Commerce/Hooks.php [hunk 4]
```diff
@@ -324,16 +324,14 @@ public function do_checkout_parse_request() {
...
-	 * @param int $post_id    the post or event id for the attendee.
-	 * @param int $product_id the ticket-product id in Tribe Commerce.
+	 * @since TBD Updated method to use the new hook from Attendee class.
+	 *
+	 * @param int $attendee_id the attendee id.
 	 */
-	public function update_stock_after_deletion( $ticket_id, $post_id, $product_id ) {
-		$this->container->make( Ticket::class )->update_stock_after_deletion( $ticket_id, $post_id, $product_id );
+	public function update_stock_after_attendee_deletion( $attendee_id ) {
+		$this->container->make( Ticket::class )->update_stock_after_attendee_deletion( $attendee_id );
```
Comment (bordoni) [inline src/Tickets/Commerce/Hooks.php:334]: We can't just delete a method like that when it's public, move it to the end and deprecate it properly.

**Review by bordoni — CHANGES_REQUESTED** [review]

**Review by codingmusician — DISMISSED** [review]

File: src/Tickets/Commerce/Attendee.php [hunk 5]
```diff
@@ -314,6 +318,9 @@ public function delete( $attendee_id, $force = true ) {
 		 */
 		do_action( 'tec_tickets_commerce_attendee_after_delete', $attendee_id, $result, $force );
 
+		\Tribe__Tickets__Attendance::instance( $event_id )->increment_deleted_attendees_count();
+		\Tribe__Post_Transient::instance()->delete( $event_id, \Tribe__Tickets__Tickets::ATTENDEES_CACHE );
```
[the remark this answers is not in the document]
Comment (rafsuntaskin) [reply src/Tickets/Commerce/Attendee.php]: Moved under deletion handler.

**Review by codingmusician — APPROVED** [review]

**Pull request merged**